site/source/docs/porting/pthreads.rst

```diff
@@ -4,50 +4,58 @@
 
-- At runtime, you can `#include <emscripten/threading.h>` and use the ``emscripten_has_threading_support()`` function to
-  test whether the currently executing code was compiled with pthreads support
-  enabled. If this function returns true, then the currently executing code was
-  compiled with ``-s USE_PTHREADS=1`` (and the current browser supports
-  multithreading).
-
-  If code is compiled with ``-s USE_PTHREADS=1`` and the current browser does
-  not support multithreading, then an exception will be thrown at page load
-  time.  It is not possible to build one binary that would be able to leverage
-  multithreading when available and fall back to single threaded when not. For
-  such backwards compatibility, two separate builds must be done, one with ``-s
-  USE_PTHREADS=1`` and the other with ``-s USE_PTHREADS=0``.
-
-- When the linker flag `-s PTHREAD_POOL_SIZE=<integer>` is not specified and `pthread_create()` is called, the new thread will not actually start to run immediately, but the main JS thread must yield execution back to browser first. This behavior is a result of `#1049079 <https://bugzilla.mozilla.org/show_bug.cgi?id=1049079>`_.
+- When the linker flag ``-s PTHREAD_POOL_SIZE=<integer>`` is not specified and ``pthread_create()`` is called, the new thread will not actually start to run immediately, but you must yield execution back to the browser's main event loop first (in order for ``new Worker`` and the related events to happen). If you don't wait but synchronously expect it to be ready (like if you join on it, or if you look at memory where you expect it to write) things may not work. Using a pool allows things to work more like a native build.
 
-- Currently several of the functions in the C runtime, such as filesystem functions like `fopen()`, `fread()`, `printf()`, `fprintf()` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via `malloc()` and `free()` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
+- Currently several of the functions in the C runtime, such as filesystem functions like ``fopen()``, ``fread()``, ``printf()``, ``fprintf()`` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via ``malloc()`` and ``free()`` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
```

Comment:
"main application thread" should be "main browser thread" right? This happens in both PROXY_TO_PTHREAD and not?

Reply:
Is this an exhaustive list of proxied functions? We say "like" and "etc" but what does that mean? How are people supposed to reason about this and figure out whether another function might also be proxied? If this can cause deadlocks or other unexpected behavior we should be much clearer about it.

Reply:
I think this is just true for things like the filesystem and DOM access, which depend on JS and so must be singlethreaded. Also we can only print the output on the browser UI thread (yes, it's the main browser thread here, in all modes). I'll clarify the sentence to mention just filesystem stuff.